iwyu.cc

```diff
@@ -2553,6 +2553,20 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     // forward-declarable, even if we're part of a pointer type, or in
     // a template argument, or whatever.
     current_ast_node()->set_in_forward_declare_context(false);
+
+    // When a method is defined outside of a class it has nns as qualifier.
+    // But for method call it doesn't matter if method is defined inside
+    // a class or outside of it.  Skip traversing NestedNameSpecifier
+    // in this case.
+    const ASTNode* ast_node = current_ast_node();
```

Comment:
I was playing with that but `set_in_forward_declare_context` is mutating and doesn't work with `const`. I still want to use const for querying ancestors and current approach seems to be short and avoiding strange naming like mutable_ast_node/ast_node or ast_node/const_ast_node.